### DISCUSSION TITLE
Are normals in the output unused?

### DISCUSSION THREAD

**Discussion by paulmelis** [MrNeRF/LichtFeld-Studio#27] [Q&A]
Just noticed that a zero-vector is written in the .ply output for each splat normal, and after checking some of the pre-trained INRIA models and their code found they also write zero-vectors?

Edit: I guess the rotation values are used to align the splats, with the nx/ny/nz fields written to the .ply file to be easier to read in general 3D packages. But still takes up unnecessary space in the file.

**Comment by MrNeRF** [comment]
Yeah, colmap is not providing any normals as output. Compare: https://github.com/colmap/colmap/blob/1f69517d360463bf2e8936fdd1f1e61f719ea6eb/src/colmap/scene/reconstruction.cc#L1991-L2013

How should be the normals calculated? There are no triangles where you could easily infer the information. It is a point cloud so where should the normal point? Correct me if I am wrong.

**Reply by paulmelis** [reply]
Well, since the splats are conceptually similar to oriented discs you could think of them as having a normal after training (I was not referring to the input point cloud from COLMAP).

**Reply by MrNeRF** [reply]
I can only guess but they are not needed for the training process. Maybe that's why they are not computed. 
Are you familiar with the gaussian splatting literature in general and how this technique was applied before? I think there are some references in the paper but I didn't refer to them. At one point I should teach myself more background how this was used before.

**Reply by paulmelis** [reply]
No, the paper is actually a bit sparse on Gaussians and glances over quite a few things. The two Zwicker references have most of the background math and concepts it seems. The 3DGS paper might have won Best Paper at SIGGRAPH for its research, the paper itself is slightly less approachable :)

**Reply by MrNeRF** [reply]
I envision this as a long-term project, so there's much to discover. It might also be beneficial to delve into the Zwicker paper. I anticipate a surge of extensions around Christmas or early 2024. Exciting! We will see where this will head.
